apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsObjectControl.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  Generate,
+  findUISchema,
+  isObjectControl,
+  rankWith,
+  type RankedTester,
+  type StatePropsOfControlWithDetail,
+} from '@jsonforms/core'
+import { JsonFormsDispatch, withJsonFormsDetailProps } from '@jsonforms/react'
+import { useMemo } from 'react'
+import { JSON_FORMS_RANKING } from '~/constants/formBuilder'
+
+export const jsonFormsObjectControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.ObjectControl,
+  isObjectControl,
+)
+
+export function JsonFormsObjectControl({
+  path,
+  visible,
+  renderers,
+  cells,
+  schema,
+  enabled,
+  uischema,
+  uischemas,
+  rootSchema,
+}: StatePropsOfControlWithDetail) {
+  const detailUiSchema = useMemo(
+    () =>
+      findUISchema(
+        uischemas || [],
+        schema,
+        uischema.scope,
+        path,
+        () =>
+          Generate.uiSchema(schema, 'VerticalLayout', undefined, rootSchema),
+        uischema,
+        rootSchema,
+      ),
```

Comment:
minor nit because i understand this is same as the docs but in here, we're saying that we want the caller to make the determination of the UI schema but i thought actually that should be the responsibility of the component?

not too important to fix since i suspect we'll only have 1 call site

Reply:
Hmm the UI schema is entirely generated from the original JSON schema, so the responsibility lies within the caller to structure the JSON schema in such a way that influences how the form should look like (e.g. radio vs dropdown, stuff like that).

Reply:
gotcha, thanks!